pennylane/operation.py

```diff
@@ -1094,6 +1094,28 @@ def queue(self, context=qml.QueuingContext):
         context.append(self)
         return self  # so pre-constructed Observable instances can be queued and returned in a single statement
 
+    def __add__(self, other):
+        r"""The addition operation between operators."""
+        return qml.ops.math.Sum(self, other)
+
+    def __sub__(self, other):
+        r"""The subtraction operation between operators."""
+        return self.__add__(other.__mul__(-1))
+
+    def __mul__(self, a):
+        r"""The scalar multiplication for operators."""
```

Comment:
Hi @mariaschuld. Is this an objective of this PR to allow a multiplication of this form:

```
p1 = qml.PauliX(0) @ qml.PauliZ(1)
p2 = qml.PauliY(0) @ qml.PauliZ(1)

p1 @ p2 # which should give  -1j Z0 as the result
```

Reply:
Yes and no :) The prototype does not have this capability, but would return a `Prod` class instance with decomposition ``d = [PauliX(0) @ PauliZ(1), PauliY(0) @ PauliZ(1) ]``.

There is also a function to flatten this: ``flatten_decomposition(d)`` will return ``[PauliX(0), PauliZ(1), PauliY(0), PauliZ(1) ]``.

A step not implemented here (and this is a major open effort) is "compilation" or "simplification", which would be a function ``simplify_decomposition(d)`` that recognises the much simpler result. 

The basic idea is to implement these building blocks as functions and then decide later which one to call automatically when evaluating the representations of arithmetic classes. 


Also note: following the idea of this prototype, you wouldn't get 1j Z0 as the result of multiplication, but a symbolic class instance that "tracks" or remembers the full multiplication. You can then ask this class for a decomposition...